app/models.py

```diff
@@ -2038,18 +2034,19 @@ class DateTimeDimension(db.Model):
 class FactNotificationStatus(db.Model):
     __tablename__ = 'ft_notification_status'
 
-    bst_date = db.Column(db.Date, index=True, primary_key=True, nullable=False)
-    template_id = db.Column(UUID(as_uuid=True), primary_key=True, index=True, nullable=False)
+    bst_date = db.Column(db.Date, index=True, primary_key=True, nullable=False, default=datetime.datetime.utcnow)
+    template_id = db.Column(UUID(as_uuid=True), primary_key=True, index=True, nullable=False, default=uuid.uuid4)
     service_id = db.Column(
         UUID(as_uuid=True),
         primary_key=True,
         index=True,
         nullable=False,
+        default=uuid.uuid4,
     )
-    job_id = db.Column(UUID(as_uuid=True), primary_key=True, index=True, nullable=False)
-    notification_type = db.Column(db.Text, primary_key=True, nullable=False)
-    key_type = db.Column(db.Text, primary_key=True, nullable=False)
-    notification_status = db.Column(db.Text, primary_key=True, nullable=False)
+    job_id = db.Column(UUID(as_uuid=True), primary_key=True, index=True, nullable=False, default=uuid.uuid4)
+    notification_type = db.Column(db.Text, primary_key=True, nullable=False, default=SMS_TYPE)
+    key_type = db.Column(db.Text, primary_key=True, nullable=False, default=KEY_TYPE_NORMAL)
+    notification_status = db.Column(db.Text, primary_key=True, nullable=False, default=NOTIFICATION_CREATED)
```

Comment:
Actually, why do we need any defaults? Wouldn't this indicate a failure somehow? Such that an email would default to sms? or should the defaults indicate that something unexpected happened?

Reply:
Right now those fields are part of the primary key and have no defaults, so all of our code is explicitly setting them.  In the future, new code could create these models and the primary key fields will have some sane defaults.  I think the only benefit is convenience, and SQL Alchemy wants to be told explicitly what to do with primary key fields.

Reply:
"In the future" seems out of scope of the work for this ticket, unless setting a default clears those SAWarnings. 

I fear the unintended consequences of setting defaults. I worked with a DBA once who insisted on defaults, but he also used a unique sets of characters so that it was obvious a default was used. I suggest removing all the defaults that were added and we have a team discussion about this change. 

Again, if this change is necessary to clear SAWarnings, I'll approve, but I need something more than "in the future" please!

Reply:
@cris-oddball I made similar comments, but I think Michael is correct that this is necessary to appease SQLAlchemy.

Reply:
@kalbfled documentation?

Reply:
I'm just going off of the text of the warning message, which warns that a primary key doesn't have a default value.

Reply:
I think this is resolved now, are there any other concerns?

Reply:
Understood and resolving comment.